### ISSUE TITLE
use-package doesn't load pinned package from package.el

### ISSUE THREAD

**Issue by davendesai** [jwiegley/use-package#224]
I have the following code in my init.el:

```lisp
(require 'package)
(setq package-enable-at-startup nil)
(setq package-archives '(("melpa" . "http://melpa.org/packages/")
  ("org" . "http://orgmode.org/elpa/")))
(package-initialize)

;; Bootstrap `use-package'
(unless (package-installed-p 'use-package)
  (package-refresh-contents)
  (package-install 'use-package))
(require 'use-package)

;; Bootstrap `org-mode'
(use-package org
  :ensure t
  :pin org)
```

Despite pinning the org package to use the corresponding repo, which I have verified appears in the package-list, use-package seems to be OK with the builtin.

**Comment by thomasf** [comment]
should work as in works for me.. I explicitly install org with `:ensure org-plus-contrib` but the `org` one gets pulled as a dependency, probably.

**Comment by thomasf** [comment]
Maybe you might want to wrap the package stuff inside a `eval-and-compile` form but that should not be the problem unless it only happens when you compile your init.

**Comment by davendesai** [comment]
Hmmm that's strange. I have also tried it with a clean install by deleting the entire ```elpa``` directory as well without results.

I've gone ahead and used the ```org-plus-contrib``` package as suggested and it seems to have worked by pulling from the correct repo. 

The documentation on [orgmode.org](http://orgmode.org/elpa.html) suggests that the ```org``` package is already up-to-date with the builtin. Maybe that's why use-package seems to gloss over without a message.

**Comment by thomasf** [comment]
Also you must have emase 24.4 or later because pinning was introduced in that version.

**Comment by davendesai** [comment]
I tapped the homebrew formula ```railwaycat/emacs-mac-port``` and installed the package ```emacs-mac```. My current version is 24.5 and ```use-package``` is up-to-date as well.

**Comment by thomasf** [comment]
Hmm.. I see that I also have a package supposedly pinned to melpa-stable but it's fetched from regular melpa anyway...

**Comment by kovrik** [comment]
@davendesai 
Should be fixed now by #302 
Can you check it please?
